qiita_plugins/biom_type/biom_type_plugin/summary.py

```diff
@@ -31,10 +44,79 @@ def generate_html_summary(qclient, job_id, parameters, out_dir):
     ValueError
         If there is any error gathering the information from the server
     """
-    # Only one paramater
-    # artifact_id = parameters['artifact']
-    # TODO
     # Step 1: gather file information from qiita using REST api
+    artifact_id = parameters['input_data']
+    qclient_url = "/qiita_db/artifacts/%s/filepaths/" % artifact_id
+    fps_info = qclient.get(qclient_url)
+    if not fps_info or not fps_info['success']:
+        error_msg = "Could not get artifact filepath information: %s"
+        if fps_info:
+            error_msg = error_msg % fps_info['error']
+        else:
+            error_msg = error_msg % "could not connect with the server"
+        raise ValueError(error_msg)
+
+    # if we get to this point of the code we are sure that this is a biom file
+    # and that it only has one element
+    fps, fps_type = fps_info['filepaths'][0]
+
     # Step 2: generate HTML summary
+    # Modified from https://goo.gl/cUVHgB
+    biom = load_table(fps)
+    num_features, num_samples = biom.shape
+
+    sample_count_summary, sample_counts = count_summary(biom, axis='sample')
+    ax = sns.distplot(sample_counts)
+    ax.set_xlabel("Number of sequences per sample")
+    ax.set_ylabel("Frequency")
+    plot = ax.get_figure()
+    sc_plot = StringIO()
+    plot.savefig(sc_plot, format='png')
+    sc_plot.seek(0)
+
+    uri = 'data:image/png;base64,' + quote(b64encode(sc_plot.buf))
+    artifact_information = [
+        "<b>Number of samples:</b> %d<br/>" % num_samples,
```

Comment:
Not sure, I guess it depends on what you like ... as you can imagine I like it like this.